### ISSUE TITLE
Caching false-y data

### ISSUE THREAD

**Issue by cvshumake** [cvshumake/CustomDB#8]
My returns from cache is the value itself. If the value does not exist/is falsy, I'm returning that falsy value. 
It allows me to cache those values. It also means that I have to check the cache first to see that the key is set.
No checking the key is set (race condition: having the key disappear between the check and get) means I'm throwing an exception.
AFAIK, I can't thread any access to the cache as its written now without implementing locking (no) or just accepting that exceptions will be part of the normal code path.
Dig into it for awhile, figure it out before writing too much client-code.

In summary, choose one:
1) Inability to cache False
2) "Atomic" cache hits
3) Exceptions become a normal part of the code path
4) Research until none of these are necessary.
5) All cache returns come back wrapped in a cache related object/array. Note that it would then be decoratable.
